src/FactorRotations.jl

```diff
@@ -62,6 +61,26 @@ function setverbosity!(verbose::Bool)
     return nothing
 end
 
+struct AutodiffBackend{B}
+    AutodiffBackend(backend::Symbol) = new{backend}()
+end
+
+const AUTODIFF_BACKEND = Ref{AutodiffBackend}(AutodiffBackend(:Enzyme))
+
+"""
+    set_autodiff_backend(backend::Symbol)
+
+Sets the *automatic differentiation* backend.
+
+Automatic differentiation is used by the fallback `criterion_and_gradient!()` implementation.
+Currently, only `:Enzyme` backend is supported.
+"""
+function set_autodiff_backend(backend::Symbol)
```

Comment:
This should probably be a bang function (`set_autodiff_backend!`) since it modifies a global variable

Reply:
I can make it a "bang" function, but the setter name already suggests that it modifies the global state.
In Julia Base these functions don't have "!", e.g. `BLAS.set_num_threads()`